docs/IPLD-ProllyTree-Spec.md

```diff
@@ -0,0 +1,271 @@
+### IPLD
+IPLD is the data model of the content-addressable web. It allows us to treat all hash-linked data structures as subsets of a unified information space, unifying all data models that link data with hashes as instances of IPLD. In the project(IPLD ProllyTree) we use [IPLD node](https://ipld.io/docs/data-model/node/)(bindnode), [IPLD schema](https://ipld.io/docs/schemas/) and [IPLD ADL](https://ipld.io/docs/advanced-data-layouts/)(Advanced Data Layouts)
+
+## Architecture
+
+![img6](./images/spec/img_6.png)
+
+Core structure of (IPLD)ProllyTree is the ProllyTree.It includes the vital fields and api to use such as Get, Search, Delete…Some of them don’t update the tree, but some will do that and rebuild the tree. ProllyTree does not store the total tree in the structure, it only saves the root node.Reading the relative tree path from nodestore while some traversal action is happening
+
+![img7](./images/spec/img_7.png)
+
+Framework is used for tree building. We input sorted data into the framework then call the BuildTree to get a new ProllyTree from the data.The tree is built level by level(bottom) by LevelBuilders. Leaf nodes are linked by parent nodes by the cid link(IPLD node). All nodes are saved in NodeStore referenced by cid.
+
+![img8](./images/spec/img_8.png)
+
+IPLD ProllyTree implemented the IPLD ADL(Advanced Layout) interface based on above structures.It includes ProllyTreeADLPrototype, Builder, TreeAssembler(MapAssembler) and ADL Node. It helps people who want to build large map(dict) by ProllyTree with ADL in IPLD world.
+
+## Modules
+### Storge(NodeStore)
+
+```golang
+type NodeStore interface {
+	WriteNode(ctx context.Context, nd *ProllyNode, prefix *cid.Prefix) (cid.Cid, error)
+	ReadNode(ctx context.Context, c cid.Cid) (*ProllyNode, error)
+
+	WriteTree(ctx context.Context, tree *ProllyTree, prefix *cid.Prefix) (cid.Cid, error)
+	ReadTree(ctx context.Context, c cid.Cid) (*ProllyTree, error)
+
+	WriteTreeConfig(ctx context.Context, cfg *TreeConfig, prefix *cid.Prefix) (cid.Cid, error)
+	ReadTreeConfig(ctx context.Context, c cid.Cid) (*TreeConfig, error)
+
+	WriteProof(ctx context.Context, prf Proof, prefix *cid.Prefix) (cid.Cid, error)
+	ReadProof(ctx context.Context, c cid.Cid) (Proof, error)
+
+	LinkSystem() *ipld.LinkSystem
+
+	Close()
+}
+```
+
+NodeStore is an import module that saves almost all information and data of ProllyTree. It is used anywhere in the project such as tree building, updating, getting…… We can also read the saved tree before. The nodestore we use in the project is based on the blockstore(ipfs)
```

Comment:
```suggestion
NodeStore is an important module that saves almost all information and data of ProllyTree. It is used anywhere in the project such as tree building, updating, getting, etc. It can also read previously saved tree nodes. The nodestore we use in the project is based on the IPFS blockstore interface.
```